### ISSUE TITLE
Feature request: enable running 'carton install' withouth '--notest'

### ISSUE THREAD

**Issue by ureesoriano** [perl-carton/carton#205]
Carton does install modules with '--notest' by default.
Would it be possible to provide any mechanism to prevent that? (EG: an env var?)

Doing so would make it possible to send carton install logs to CPAN Testers.

**Comment by maxhq** [comment]
FYI: There is an aged pull request #145 that adds a --with-test option...

**Comment by karenetheridge** [comment]
Yes please -- being able to test the dependencies is very important for incremental build systems like travis, and without this option one needs to build a separate test instance that uses `cpanm` directly on the cpanfile (which will bypass cpanfile.snapshot, thereby testing different versions of prerequisite modules)

**Comment by miyagawa** [comment]
just to update my thoughts on this - i think there should be a command to test your module (or any perl code, really) _against_ the bundle of the modules you currently have. Rather than testing it _during_ the build & install, which might pass or fail depending on the order of installation, it would be better to have a command to run tests against the set of the modules, after the fact.

Unfortunately I don't have the working code that does exactly that, but it might not be too hard to do it because once you finish `carton install` you get all the libraries in `./local/lib/perl5`, all you have to do is to visit the dependencies and run the tests with the proper `PERL5LIB` variable.

**Comment by KES777** [comment]
Running tests during build&install is the only way to test every module. I can not imagine how to get list of all modules `after the fact`.

`carton` is the best program to do that. Why just not allow people to pass those flags to underlied program which they want?

If hypothetically `carton` will support `--with-test` option, how it will harm? If it will not harm then I do not see reasons to not support it.
